### ISSUE TITLE
Build failure with GHC 8.4

### ISSUE THREAD

**Issue by DanBurton** [conal/TypeCompose#6]
Add semigroup instances.
See also: https://ghc.haskell.org/trac/ghc/wiki/Migration/8.4#SemigroupMonoidsuperclasses

```
Building library for TypeCompose-0.9.12..

[ 3 of 10] Compiling Control.Compose  ( src/Control/Compose.hs, dist/build/Control/Compose.o )

src/Control/Compose.hs:596:10: error:
    • Could not deduce (Semigroup (Flip j o a))
        arising from the superclasses of an instance declaration
      from the context: (Applicative (j a), Monoid o)
        bound by the instance declaration
        at src/Control/Compose.hs:596:10-61
    • In the instance declaration for ‘Monoid (Flip j o a)’
    |
596 | instance (Applicative (j a), Monoid o) => Monoid (Flip j o a) where
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Control/Compose.hs:645:10: error:
    • Could not deduce (Semigroup (App f m))
        arising from the superclasses of an instance declaration
      from the context: (Applicative f, Monoid m)
        bound by the instance declaration
        at src/Control/Compose.hs:645:10-54
    • In the instance declaration for ‘Monoid (App f m)’
    |
645 | instance (Applicative f, Monoid m) => Monoid (App f m) where
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Control/Compose.hs:851:1: error:
    • Could not deduce (Semigroup (Arrw j f g a))
        arising from the superclasses of an instance declaration
      from the context: Monoid (j (f a) (g a))
        bound by the instance declaration
        at src/Control/Compose.hs:851:1-63
    • In the instance declaration for ‘Monoid (Arrw j f g a)’
    |
851 | deriving instance Monoid (f a `j` g a) => Monoid (Arrw j f g a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by guibou** [comment]
AFAIK it was closed in #7

**Comment by conal** [comment]
I just released 0.9.13 to Hackage with the `Semigroup` fix from @guibou . Sorry about the delay.
